src/psyclone/psyir/symbols/symbol_table.py

```diff
@@ -1566,6 +1681,183 @@ def copy_external_import(self, imported_var, tag=None):
                         f"Couldn't copy '{imported_var}' into the SymbolTable."
                         f" The tag '{tag}' is already used by another symbol.")
 
+    def _import_symbols_from(self, csymbol: ContainerSymbol,
+                             container,
+                             symbol_target: Optional[Symbol] = None
+                             ) -> Set[Symbol]:
+        '''
+        Imports symbols from the supplied Container into this table. If
+        `target_symbol` is specified then only that symbol is imported.
+        Otherwise, this method imports all symbols named in the import
+        (i.e. if it corresponds to a USE xx, Only: yy) or, if it
+        is a wildcard import, all public symbols in the source Container.
+
+        :param csymbol: the ContainerSymbol from which to import.
+        :param container: the PSyIR of the Container from which to import.
+        :type container: :py:class:`psyclone.psyir.nodes.ContainerNode`
+        :param symbol_target: optional, single symbol to attempt to import.
+
+        :returns: the Symbols that have been added to this table.
+
+        '''
+        # Set holding all Symbols that we succeed in importing.
+        imported_names = {}
+        # Map from symbol name in source container to name at import site.
```

Comment:
The descriptions are swapped? Also can you rename "imported_names" to "qualified_imported_names"

Reply:
Oh yes, and done.